pkg/backend/token_authcode.go

```diff
@@ -41,29 +38,41 @@ type refreshDescriptor struct {
 var _ scheduler.Descriptor = &refreshDescriptor{}
 
 func (rd *refreshDescriptor) Run(ctx context.Context, pc chan<- scheduler.Process) error {
-	ticker := rd.backend.clock.NewTicker(refreshInterval)
-	defer ticker.Stop()
+	// start out checking once per minute
+	refreshInterval := 60
+	ticker := rd.backend.clock.NewTicker(time.Duration(refreshInterval) * time.Second)
 
 	for {
-		err := rd.backend.data.Managers(rd.storage).AuthCode().ForEachAuthCodeKey(ctx, func(keyer persistence.AuthCodeKeyer) {
-			proc := &refreshProcess{
-				backend: rd.backend,
-				storage: rd.storage,
-				keyer:   keyer,
+		c, err := rd.backend.getCache(ctx, rd.storage)
+		if err == nil && c != nil && c.Config.RefreshInterval != 0 {
```

Comment:
This check should probably be `> 0`, not `!= 0`. I don't know what will happen if you create a ticker with a negative interval, but I bet it won't be good. We also need an upper bound check against `time.Duration(math.MaxInt64)/time.Second` so that we don't wrap around. (This could happen here or where the config is written out -- I'm ambivalent.)

Reply:
Ok on `> 0`.  I don't understand the need for an upper bound check though.  How might the lack of a check cause a problem?

Reply:
If you pick a _very_ large value, anything larger than 9223372036 (which is the value of `math.MaxInt64 / time.Second`), then when you later multiply the value by time.Second to get a time.Duration you end up with integer overflow and effectively a negative duration.

https://play.golang.org/p/Eyy6trmBzCg

Reply:
Ok, then they would get a negative number which would disable the timer, but would anybody tell the difference between that and 9223372037 seconds which is over 250 years?  It's not going to crash (I tried it) so I don't see the need to protect against such an unreasonable setting.

Reply:
Generally speaking, protecting against integer overflows is pretty much always a good idea. For example, even though it works now, if it's technically "undefined" per Go's docs, it could break in the future. But your point is taken that it's sort of a crazy thing to set to begin with. I'm trying to think, if you don't want to outright disable it, what a sane maximum is. Maybe 90 days? I would be also OK with just setting a hard limit of something reasonable like that.

Reply:
Ok I put in a check in path_config.go for less than zero or greater than 90 days.